docs/bsp.md

```diff
@@ -251,24 +251,22 @@ trait BuildTargetIdentifer {
 
 ### Task Id 
 
-The Task Id allows clients to uniquely identify a resource and establish a client-parent
-relationship with another id.
+The Task Id allows clients to *uniquely* identify a BSP task and establish a client-parent
+relationship with another task id.
 
 ```scala
 trait TaskId {
-  /** The id */
+  /** The id, it has to be unique. */
   def id: String
```

Comment:
I've been tempted to make this a `URI` instead of a `String`, but I'm not sure it's the way to go. From the build tool side, it seems simpler to me to create a unique ID with a `URI` than creating an hexadecimal random string of a concrete length.
  
What do others think about this?

Reply:
Proposed reformulation

> A unique identifier

Reply:
@jvican a random unique string is straightforward (eg random UUID or or even just random long), transforming it to a URI is an additional burden and not necessary